### PR TITLE
Enable CI

### PR DESCRIPTION
**Pull request by perdy**
Create two CI pipelines for feature branches and master branch that consists of following steps:

1. Test.
    * Code format check (black).
    * Imports order check (isort).
    * Code lint (flake8).
    * Run Tests.
2. Release a new version to PyPI (only for commits in master associated with a new version tag)

### CODE REVIEW

File: .github/workflows/ci_master.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v[0-9]*.[0-9]*.[0-9]*"
```
Comment (antogc) [inline .github/workflows/ci_master.yaml:8]: Does that push action covers when a pull-request is finally merged?

File: .github/workflows/ci_master.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        run: python make flake8
+      - id: tests
+        name: Tests
+        run: python make test
+  release:
+    needs: test
+    name: Release a new version
+    runs-on: ubuntu-latest
+    container:
+      image: perdy/github-actions-builder:latest
```
Comment (antogc) [inline .github/workflows/ci_master.yaml:43]: Shouldn't we have that image in pikselpalette space?

**Review by antogc — APPROVED** [review]

**Review by antogc — APPROVED** [review]

**Review by antogc — APPROVED** [review]

**Review by antogc — APPROVED** [review]

File: .github/workflows/ci_master.yaml (hunk 2 above)
Comment (perdy) [reply .github/workflows/ci_master.yaml:43]: We could fork that image into Piksel space in a future, but for now I'd say it's fine, this image is open sourced so anyone that need a change could request it.

File: .github/workflows/ci_master.yaml (hunk 1 above)
Comment (perdy) [reply .github/workflows/ci_master.yaml:8]: The process for a PR is: Create PR > Check in that branch > Accept PR and Merge > Check in master branch; so yes, it should cover when a PR is merged.

**Review by antogc — APPROVED** [review]

**Review by antogc — APPROVED** [review]

**Review by antogc — APPROVED** [review]

**Review by antogc — APPROVED** [review]

**Review by antogc — APPROVED** [review]
Approved

**Pull request merged**